crates/sargon/src/system/sargon_os/pre_authorization/sargon_os_pre_authorization_status.rs

```diff
@@ -0,0 +1,392 @@
+use crate::prelude::*;
+use std::time::Duration;
+
+// ==================
+// Poll PreAuthorization Status (Public)
+// ==================
+impl SargonOS {
+    /// Polls the state of a PreAuthorization until we can determine its `TransactionStatus`.
+    pub async fn poll_pre_authorization_status(
+        &self,
+        intent_hash: SubintentHash,
+        expiration: Option<DappToWalletInteractionSubintentExpiration>,
+    ) -> Result<TransactionStatus> {
+        let (status, _) = self
+            .poll_subintent_status_with_delays(intent_hash, expiration)
+            .await?;
+
+        Ok(status)
+    }
+}
```

Comment:
This function is not very useful IMO, unless you plan to extend it

Reply:
We need an internal fn that also returns the delays to test it. If we are going to use this internal one (`poll_pre_authorization_status_with_delays`) from UniFFI, the `into_result()` conversion wouldn't be as straightforward right?